src/targets/maven.ts

```diff
@@ -0,0 +1,389 @@
+};
+
+/**
+ * Target responsible for uploading files to Maven Central.
+ */
+export class MavenTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'maven';
+  /** Target options */
+  public readonly mavenConfig: MavenTargetConfig;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.mavenConfig = this.getMavenConfig();
+    this.checkRequiredSoftware();
+    this.makeRegexpFromConfiguration();
+  }
+
+  /**
+   * Returns the maven config with the required data (e.g. environment
+   * variables) for this target. If there's a configuration requirement missing,
+   * raises an error.
+   *
+   * @returns the maven config for this target.
+   */
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      ...this.getTargetSecrets(),
+      ...this.getTargetSettings(),
+    };
+  }
+
+  private getTargetSecrets(): Record<TargetSettingType, string> {
+    const secrets = targetSecrets.map(secret => {
+      checkEnvForPrerequisite({ name: secret });
+      return {
+        name: secret,
+        value: process.env[secret],
+      };
+    });
+    return this.reduceConfig(secrets);
+  }
+
+  private reduceConfig(
+    config: { name: string; value: string | undefined }[]
+  ): Record<string, string> {
+    return config.reduce((prev, current) => {
+      return {
+        ...prev,
+        [current.name]: current.value,
+      };
+    }, {});
+  }
+
+  private getTargetSettings(): Record<TargetSettingType, string> {
+    const settings = targetOptions.map(setting => {
+      return {
+        name: setting,
+        value: this.config[setting],
+      };
+    });
+    return this.reduceConfig(settings);
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   * If there's required software missing, raises an error.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug(
+      `Checking if Maven CLI is available on ${this.mavenConfig.mavenCliPath}...`
+    );
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
+    this.logger.debug(
+      `Checking if Gradle CLI is available on ${this.mavenConfig.gradleCliPath}...`
+    );
+    checkExecutableIsPresent(this.mavenConfig.gradleCliPath);
+    this.logger.debug(`Checking if GPG is available in the path...`);
+    checkExecutableIsPresent('gpg');
+  }
+
+  /**
+   * Parses the required target's configuration variables' types from strings to
+   * RegExp.
+   *
+   * To make TS happy, they must be `RegExp` objects. It's not valid to write
+   * `/myRegex/`
+   */
+  private makeRegexpFromConfiguration(): void {
+    this.config.androidDistDirPattern = new RegExp(
+      this.config.androidDistDirPattern
+    );
+    this.config.androidFileSearchPattern = new RegExp(
+      this.config.androidFileSearchPattern
+    );
+  }
```

Comment:
Maybe you can use this:

https://github.com/getsentry/craft/blob/150310a11c4896e86df8ca26ef36e10ca5a44c8d/src/utils/filters.ts#L17-L26